mex/common/identity/query.py

```diff
@@ -0,0 +1,43 @@
+from mex.common.exceptions import MExError
+from mex.common.identity.dummy import DummyIdentityProvider
+from mex.common.identity.models import Identity
+from mex.common.identity.types import IdentityProvider
+from mex.common.settings import BaseSettings
+from mex.common.types import Identifier
+
+
+def upsert_identity(
+    had_primary_source: Identifier,
+    identifier_in_primary_source: str,
+    stable_target_id: Identifier,
+    entity_type: str,
+) -> Identity:
+    """Insert a new identity or update an existing one."""
+    settings = BaseSettings.get()
+    if settings.identity_provider == IdentityProvider.DUMMY:
```

Comment:
If identity provider is not dummy, then request will not go through here, right?

Reply:
yes, correct. when we add the backend-identity-provider as a second option we can easily integrate it here though 🙂